### PR TITLE
chore: treefmt formatting

### PR DESCRIPTION
**Pull request by blaggacao**
- chore: enable shfmt and prettier with treefmt
- style: treefmt
- chore: ignore formatting commit in git blame

# Context

Same as #825, but with ignoring the formatting commit during blame.

# Proposed Solution

- Isolates the reformatting into a single commit based on an already fine-tuned `treefmt.toml` (from #825)
- Adds that commit to the ingore rev for blame file

### CODE REVIEW

File: .git-blame-ignore-revs [hunk 1]
```diff
@@ -0,0 +1,4 @@
+# To apply this file locally, run: `git config blame.ignoreRevsFile .git-blame-ignore-revs`
+
+# applied treefmt for the first time
+3391af283a72871b3afe23dd721e35c52daba241
```
Comment (gytis-ivaskevicius) [inline .git-blame-ignore-revs:4]: Not sure if it's all that helpful and probably not worth keeping it around. CC @rhyslbw

Comment (blaggacao) [reply .git-blame-ignore-revs:4]: This is common recommended practice and leverages the corresponding [GitHub feature](https://github.blog/changelog/2022-03-24-ignore-commits-in-the-blame-view-beta/).

Comment (rhyslbw) [reply .git-blame-ignore-revs:4]: Oh that's a cool feature! Looks good here

**Comment by blaggacao** [comment]
Adopted into #825 after consultation with @tshaynik .

**Pull request closed without merging**